File: PruebaTecnica.WebApi.DataAccess/Models/Cliente.cs
using PruebaTecnica.WebApi.DataAccess.Core.Model;

namespace PruebaTecnica.WebApi.DataAccess.Models
{
    public class Cliente : BaseModel
    {
        public string Nombre { get; set; }
        public string Apellido { get; set; }
        public System.DateTime FechaDeNacimiento { get; set; }
        public int IdGenero { get; set; }
        public int IdTipoDeIdentificacion { get; set; }
        public string NumeroDeIdentificacion { get; set; }
        public bool Maneja { get; set; }
        public bool UsaLentes { get; set; }
        public bool TieneDiabetes { get; set; }
        public bool TieneOtrasEnfermedades { get; set; }
        public string OtrasEnfermedades { get; set; }
        public bool ActivoActualmente { get; set; }
    }
}

File: PruebaTecnica.Frontend/src/app/Clientes/Models/Cliente.ts
import { IBaseEntity } from 'src/app/Common/Interfaces/IBaseEntity';
import { ISelector } from 'src/app/Common/Interfaces/ISelector';

export interface IClienteEnCRUDModel extends IBaseEntity {
  Nombre: string;
  Apellido: string;
  FechaDeNacimiento: string;
  IdGenero: number;
  Generos: ISelector[];
  IdTipoDeIdentificacion: string;
  TiposDeIdentificacion: ISelector[];
  NumeroDeIdentificacion: string;
  Maneja: boolean;
  UsaLentes: boolean;
  TieneDiabetes: boolean;
  TieneOtrasEnfermedades: boolean;
  OtrasEnfermedades: string;
  ActivoActualmente: boolean;
}

export interface IClienteEnListaModel extends IBaseEntity {
  Nombre: string;
  Apellido: string;
  Edad: string;
  Genero: string;
  TipoDeIdentificacion: string;
  NumeroDeIdentificacion: string;
  Maneja: string;
  UsaLentes: string;
  TieneDiabetes: string;
  TieneOtrasEnfermedades: string;
  OtrasEnfermedades: string;
  ActivoActualmente: string;
}
